File: apollo/src/main/java/io/muun/apollo/domain/model/base/PersistentModel.java
package io.muun.apollo.domain.model.base;

import android.support.annotation.Nullable;

public class PersistentModel {

    @Nullable
    public Long id;

    protected PersistentModel(@Nullable Long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object other) {

        if (this == other) {
            return true;
        }

        if (!(other instanceof PersistentModel)) {
            return false;
        }

        return id != null && id.equals(((PersistentModel) other).id);
    }

    @Override
    public int hashCode() {

        return id != null ? id.hashCode() : 0;
    }
}

File: android/apollo/src/main/java/io/muun/apollo/domain/model/base/PersistentModel.kt
package io.muun.apollo.domain.model.base

open class PersistentModel protected constructor(open var id: Long?) {

    override fun equals(other: Any?): Boolean {

        if (this === other) {
            return true
        }

        return if (other !is PersistentModel) {
            false
        } else id != null && id == other.id

    }

    override fun hashCode(): Int {
        return if (id != null) id!!.hashCode() else 0
    }
}
